Clearer enable and disable notifications

Ell1ott opened the pull request:
Adds custom icons to the enable and disable notifications of modules to make it easier to see if you have toggled them off or on.

Ell1ott committed:
Update theme

Ell1ott committed:
Update theme

Ell1ott committed:
white design

Ell1ott committed:
Update theme

Ell1ott committed:
Update theme

Ell1ott committed:
Update theme

Ell1ott committed:
Update theme

Ell1ott committed:
Update theme

Ell1ott committed:
added moduleS

Ell1ott committed:
Update theme

Commit:
Update ModuleAimbot.kt

Commit:
Merge branch 'nextgen' of https://github.com/CCBlueX/LiquidBounce into nextgen

Commit:
Merge branch 'nextgen' of https://github.com/Ell1ott/LiquidBounce into nextgen

Commit:
Update theme

Commit:
Merge branch 'nextgen' of https://github.com/Ell1ott/LiquidBounce into nextgen

Commit:
Update theme

Commit:
Update theme

Ell1ott committed:
fixed bow aimbot

Ell1ott committed:
Merge branch 'nextgen' of https://github.com/Ell1ott/LiquidBounce into nextgen

Ell1ott committed:
Revert "Merge branch 'nextgen' of https://github.com/Ell1ott/LiquidBounce into nextgen"

This reverts commit e4813f87b702fcf51936ec2c906386b65a2de12f, reversing
changes made to b0ccc7ae6134af8cb727abe24d97caa8d08ee7d2.

Ell1ott committed:
Merge remote-tracking branch 'upstream/nextgen' into nextgen

Ell1ott committed:
clearer enabled and disabled notifications

Ell1ott committed:
Revert "fixed bow aimbot"

This reverts commit b0ccc7ae6134af8cb727abe24d97caa8d08ee7d2.

1zun4secondary commented:
Please show a picture of the changes.

Ell1ott commented:
Certainly.
Here is a screenshot showcasing the changes:
![Screenshot 2023-06-04 171943-cut](https://github.com/CCBlueX/LiquidBounce/assets/85990359/cc3c6508-5e0e-411c-8e5b-91ff7b09620e)
and this is what it looked like before the changes:
![Screenshot 2023-06-04 172712](https://github.com/CCBlueX/LiquidBounce/assets/85990359/00e922aa-359f-4448-9dff-8e4c7f759c59)

These visual indicators are especially helpful when using a keybind to toggle a module, and you quickly want to see if you turned it on or off.

be4dev commented:
> Certainly. Here is a screenshot showcasing the changes: ![Screenshot 2023-06-04 171943-cut](https://user-images.githubusercontent.com/85990359/243188741-cc3c6508-5e0e-411c-8e5b-91ff7b09620e.png) and this is what it looked like before the changes: ![Screenshot 2023-06-04 172712](https://user-images.githubusercontent.com/85990359/243188754-00e922aa-359f-4448-9dff-8e4c7f759c59.png)
> 
> These visual indicators are especially helpful when using a keybind to toggle a module, and you quickly want to see if you turned it on or off.

well, it is still nice, but it would be better to try using icons from the concept release https://raw.githubusercontent.com/CCBlueX/LiquidBounce/nextgen/src-theme/concepts/ingame/ingame.png

Ell1ott committed:
match concept release

Ell1ott commented:
Yes, I simply overlooked that the concept release already has a solution for it, so I have some quick changes to use those instead.
Now it looks like this:
![image](https://github.com/CCBlueX/LiquidBounce/assets/85990359/e375afcd-1315-464a-8870-3908dc3aa454)

1zun4secondary commented:
Looks better, but I think there is a small problem with the concept itself. Showing an X instead of a tick makes it feel like an error. It should show a tick, but in red instead.

Ell1ott commented:
I agree that it looks like it is an error now, but I don't think just changing both to a tick is the optimal solution. this would make the tick "useless" because it wouldn't really add any more information.

Ell1ott commented:
I tried designing some new icons for it, that try to follow the concept more, but don't make it look like an error:
![image](https://github.com/CCBlueX/LiquidBounce/assets/85990359/e1c5f66a-36ef-4b1c-910d-d0b5aea7019b)
I personally like the 2 to the left the most, but let me know what you think, and if you have other ideas for possible icons for it.

1zun4secondary commented:
I like the ones to the left the most as well! Looks good and they seem to fit better into the design concept. 
Approved by @SenkJu as well.

SenkJu commented:
Left ones look good

Ell1ott commented:
Great. Then I just need to push the changes to the repo.

Ell1ott committed:
new icons

Ell1ott commented:
Everything should be set and ready to go now.

1zun4secondary commented on src/main/kotlin/net/ccbluex/liquidbounce/features/module/Module.kt:
```diff
@@ -75,7 +75,7 @@ open class Module(
                 if (new) Text.translatable("liquidbounce.generic.enabled") else Text.translatable("liquidbounce.generic.disabled"),
                 this.name,
-                NotificationEvent.Severity.INFO
+                if (new) NotificationEvent.Severity.SUCCESS else NotificationEvent.Severity.ERROR
```
This needs to be DISABLED and ENABLED now to work with the CSS name of the new icons. Please add them to the Severity Enum.

Ell1ott replied:
It does as well work currently because i added the title as another class: 
`<div
    class="icon {severity.toString().toLowerCase()} {title
      .toString()
      .toLowerCase()}"
  />` 
and then just override the icon if it as well got the class "enabled" or "disabled": 
```
&.enabled {
      background-image: url("../img/notification/enabled.svg");
    }

&.disabled {
  background-image: url("../img/notification/disabled.svg");
}
```
but adding them to the Severity Enum is definitely the cleaner and more future-proof solution, so I will change it to use it instead.

1zun4secondary replied:
Yes, especially because the title will change depending on which language you are using. 😅

Ell1ott replied:
Oh, yes, I didn't think about that. Thanks for the correction. I changed it to its own severity, so it should now be working independently of the chosen language.

![image](https://github.com/CCBlueX/LiquidBounce/assets/85990359/2aa4b02a-49d3-4365-91ed-66431d2fb409)

Ell1ott committed:
use of enabled and disabled severity

2 new severity (enabled and disabled) have been added and are being used by the notification system.

ghost commented:
If this is accepted, then it should close https://github.com/CCBlueX/LiquidBounce/issues/473.
